docs/lob_guide/src/asciidoc/_chapters/work_with_lob.adoc

```diff
@@ -0,0 +1,738 @@
+```
+
+* This example uses the `FILETOLOB` function that converts data from a local file into LOB format, and stores all data into HDFS associated with that column/row.
++
+```
+INSERT INTO tlob130txt1 VALUES(1,filetolob('lob_input_a1.txt'));
+```
+
+* This example takes an int64 value as an input which is an address to a buffer and a size parameter. The buffer contents are converted to LOB format and stored in HDFS.
++
+```
+INSERT INTO tlob1 VALUES (1, buffertolob(LOCATION 124647474, SIZE 2048));
+```
+
+* This example uses different functions to convert strings, files, external lob into LOB data. The EXTERNALTOLOB function takes an external file. 
++
+```
+INSERT INTO tlob130ext VALUES(1, STRINGTOLOB('first lob'),                                           
+FILETOLOB('hdfs:///lobs/lob_input_a1.txt'), 
+EXTERNALTOLOB('hdfs:///lobs/lob_input_a1.txt'));
+```
+
+* This example uses a parameter.
++
+```
+PREPARE S FROM INSERT INTO t130lob2 VALUES (1, ?);
+EXECUTE S USING 'fgfgfhfhfhfhhfhfhfhjfkkfkffllflflfll';
+```
+
+* This example uses the `EMPTY_BLOB` function to insert an empty lob and creates a dummy lob handle. 
++
+```
+INSERT INTO t130lob2 VALUES (1, empty_blob());
+```
+
+[#insert into a sql table containing lob columns using select clause]
+== Insert into a SQL Table Containing LOB Columns Using Select Clause
+
+[#syntax]
+=== Syntax
+
+```
+INSERT INTO target-table [(target-col-list-expr)] SELECT [source-col-list-expr] FROM source-table
+```
+
+```
+target-col-list-expr is: 
+target-colname[, target-colname]...
+
+target-colname is: 
+trafodion lob column name
+
+source-col-list-expr is: 
+source-colname[, source-colname]...
+
+source-colname is: 
+hive varchar column name          | 
+trafodion varchar column name     | 
+trafodion char column name        |
+trafodion lob column name
+
+source-table is: 
+hive table                        |
+trafodion table 
+```
+
+[#semantics]
+=== semantics
+
+* `_target-col-list-expr_`
++
+Names a single column or multiple columns enclosed in parentheses in the target table in which to insert values. 
+
++
+The data type of each target column must be compatible with the data type of its corresponding source value. 
+
++
+Within the list, each target column must have the same position as its associated source value.
+
+** `_target-colname_` 
++
+is a SQL identifier that specifies a target column name.
+
+* `_source-col-list-expr_`
++
+Names a single column or multiple columns enclosed in parentheses in the source table from which to get values. 
+
+** `_source-colname_` 
++
+is a SQL identifier that specifies a source column name.
+
+[#considerations]
+=== Considerations
+
+When inserting from a source Trafodion table column into a target table, the source column subtype of the LOB column needs to match the target table column. That is, you cannot insert from an external LOB column into a regular LOB column in the target. They both need to be the same type. 

```

Comment:
The last sentence surprises me actually. I would have thought that INSERT/SELECT might be a way to copy LOBs from external files into internal LOB values. I'm not saying this is incorrect, though (I don't know if it is). You may want to check on this.

Reply:
This restriction is correct. The source table that contains columns for an "external lob" only contains the name of the external lob file. Trafodion does not copy and store that data in our usual lob namespace - i.e user/trafodion/lobs. The descriptor files only contain the name of an hdfs file located somewhere in the system. When we do an insert-select from an external table, we copy the name of the "external" lob data file only.  So we need a place in the target table to store the external file name. This format can only be supported if the target table's column is also defined as external.  Today we allow insert-select from a hive varchar column, trafodion char or varchar column or a trafodion lob column that is not external.

Reply:
Thanks!